shopfloor_reception/services/reception.py

```diff
@@ -1241,15 +1270,35 @@ def _post_shopfloor_created_line(self, selected_line):
         )
 
     def _auto_post_line(self, selected_line):
-        new_move = selected_line.move_id.split_other_move_lines(
-            selected_line, intersection=True
+        # If user only processed 1/5 and is the only one working on the move,
+        # then selected_line is the only one related to this move.
+        # In such case, we must ensure there's another move with the remaining
+        # quantity to do, so selected_line is extracted in a new move as expected.
+
+        # Always keep the quantity todo at zero, the same is done
+        # in Odoo when move lines are created manually (setting)
+        lines_with_qty_todo = selected_line.move_id.move_line_ids.filtered(
+            lambda line: line.state not in ("cancel", "done")
+            and line.product_uom_qty > 0
         )
-        if new_move:
-            # A new move is created in case of partial quantity
-            new_move.extract_and_action_done()
-            return
-        # In case of full quantity, post the initial move
-        selected_line.move_id.extract_and_action_done()
+        if lines_with_qty_todo:
+            lines_with_qty_todo.product_uom_qty = 0
+
+        move = selected_line.move_id
+        move_quantity = move.product_uom._compute_quantity(
+            move.product_uom_qty, selected_line.product_uom_id
+        )
+        if selected_line.qty_done == move_quantity:
+            # In case of full quantity, post the initial move
+            return selected_line.move_id.extract_and_action_done()
+        split_move_vals = move._split(selected_line.qty_done)
+        new_move = move.create(split_move_vals)
+        new_move.move_line_ids = selected_line
+        new_move._action_confirm(merge=False)
+        new_move._recompute_state()
+        new_move._action_assign()
+        move._recompute_state()
```

Comment:
There should also be a lock. 
```suggestion
        lock = self._actions_for("lock")
        lock.for_update(move)
        move._recompute_state()
```

Reply:
Fixed